### PR TITLE
PTF-1948: Metrics and Logs

### PR DESCRIPTION
**Pull request by haletsky**

### CODE REVIEW

File: packages/backend/src/modules/metric/metric.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Args, Query, Resolver, Subscription } from '@nestjs/graphql';
+import { Roles } from '../../common/decorators';
+import RolesEnum from '../../common/enum/roles.enum';
+import SubscriptionEnum from '../../common/enum/subscription.enum';
```
Comment (valcker) [inline packages/backend/src/modules/metric/metric.resolver.ts:4]: Unused import

File: packages/backend/src/modules/metric/metric.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Args, Query, Resolver, Subscription } from '@nestjs/graphql';
```
Comment (valcker) [inline packages/backend/src/modules/metric/metric.resolver.ts:1]: Unused "Subscription"

File: packages/backend/src/modules/metric/metric.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { Roles } from '../../common/decorators';
+import RolesEnum from '../../common/enum/roles.enum';
+import SubscriptionEnum from '../../common/enum/subscription.enum';
+import RedisService from '../redis/redis.service';
+import MetricService from './metric.service';
+
+@Resolver('Metric')
+export default class MetricResolver {
+  public constructor(private readonly metricService: MetricService, private readonly redis: RedisService) {}
```
Comment (valcker) [inline packages/backend/src/modules/metric/metric.resolver.ts:10]: Unused `private readonly redis`. Fix ESLint errors.

File: packages/backend/src/modules/metric/metric.service.ts [hunk 4]
```diff
@@ -68,6 +71,48 @@ export default class MetricService {
...
+
+    return {
+      latency: latencyData.map((obj: any) => ({ argument: obj._id, value: obj.latency})),
+      count: latencyData.map((obj: any) => ({ argument: obj._id, value: obj.count})),
+      countries: result.countries.map((obj: any) => ({ argument: obj._id, value: obj.count }))
+    };
+  }
+
   private async getRecords(channel: MetricsChannels.REQUEST_IDS | MetricsChannels.NETWORK, chunk: number): Promise<string[]> {
     const [[error, records]] = await this.redis
```
Comment (valcker) [inline packages/backend/src/modules/metric/metric.service.ts:117]: `error` is not used.

File: packages/backend/src/modules/metric/metric.service.ts [hunk 5]
```diff
@@ -68,6 +71,48 @@ export default class MetricService {
...
+            output: { latency: { $avg: '$latency' }, count: { $sum: 1 },  },
+          } },
+        ],
+        countries: [
+          { $group: {
+            _id: '$geo.country',
+            count: { $sum: 1 },
+          } },
+        ],
+        // failures: [
```
Comment (valcker) [inline packages/backend/src/modules/metric/metric.service.ts:92]: Remove comments

File: packages/backend/src/modules/metric/metric.service.ts [hunk 6]
```diff
@@ -209,4 +254,19 @@ export default class MetricService {
       return {};
     }
   }
+
+  private getBoundries(startDate: moment.Moment, endDate: moment.Moment, scale: 'day' | 'hour' | 'week' | 'month'): Date[] {
```
Comment (valcker) [inline packages/backend/src/modules/metric/metric.service.ts:258]: Typo in the method name, it should be `getBoundaries`

Comment (valcker) [inline packages/backend/src/modules/metric/metric.service.ts:258]: Do we really need MomentJS here? It is slow, consider replacing it with native Date or with `date-fns`.

File: packages/frontend/src/presenter/Metrics/useMetrics.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    let data = { startDate: startDate, endDate: endDate, scale };
+    if (scale === 'hour') {
+      data = {
+        startDate: startDate,
+        endDate: moment(startDate).add(1, 'day').toDate(),
+        scale,
+      };
+    }
+    refetch(data);
+  }, [startDate, endDate, scale]);
```
Comment (valcker) [inline packages/frontend/src/presenter/Metrics/useMetrics.ts:23]: ESLint warnings

**Review by germankrause — APPROVED** [review]

**Pull request merged**